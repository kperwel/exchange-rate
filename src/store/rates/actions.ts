import { RatesActionTypes, FETCH_FINISHED, FETCH_ERROR, ExchangeResponse } from "./types";

export function fetchRatesError(error: Error): RatesActionTypes {
  return {
    type: FETCH_ERROR,
    payload: { error }
  };
}

export function fetchRatesFinished(response: ExchangeResponse): RatesActionTypes {
  return {
    type: FETCH_FINISHED,
    payload: { response }
  };
}
