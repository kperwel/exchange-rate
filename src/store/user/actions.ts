import { UserActionTypes, CURRENCY_CHANGE, VALUE_CHANGE, EXCHANGE } from "./types";
import { CURRENCY_TYPE } from "./reducer";

import { Transation } from "./transaction";

export function currencyChange(currency: string, currencyType: CURRENCY_TYPE): UserActionTypes {
  return {
    type: CURRENCY_CHANGE,
    payload: {
      currency,
      currencyType
    }
  };
}

export function exchange(transation: Transation): UserActionTypes {
  return {
    type: EXCHANGE,
    payload: transation
  };
}

export function valueChange(value: [string, string]): UserActionTypes {
  return {
    type: VALUE_CHANGE,
    payload: {
      value
    }
  };
}
