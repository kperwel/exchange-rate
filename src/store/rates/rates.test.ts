import * as actions from "./actions";
import userReducer, { Status } from "./reducer";

const getState = () => ({
  rates: {
    PLN: 4.2675,
    EUR: 1
  },
  date: "2019-07-10",
  baseCurrency: "EUR",
  error: null,
  status: Status.INIT
});

const response = {
  base: "PLN",
  date: "2019-07-10",
  rates: {
      EUR: 0.3
  }
};

describe("<Rates />", () => {
  it("should render finished state", () => {
    const reduced = userReducer(getState(), actions.fetchRatesFinished(response));
    expect(reduced).toMatchSnapshot();
  });

  it("should render error state", () => {
    const reduced = userReducer(getState(), actions.fetchRatesError(Error("test error")));
    expect(reduced).toMatchSnapshot();
  });
});
