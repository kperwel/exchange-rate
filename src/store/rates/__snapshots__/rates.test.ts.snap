// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rates /> should render error state 1`] = `
Object {
  "baseCurrency": "EUR",
  "date": "2019-07-10",
  "error": [Error: test error],
  "rates": Object {
    "EUR": 1,
    "PLN": 4.2675,
  },
  "status": "ERROR",
}
`;

exports[`<Rates /> should render finished state 1`] = `
Object {
  "baseCurrency": "PLN",
  "date": "2019-07-10",
  "error": null,
  "rates": Object {
    "EUR": 0.3,
    "PLN": 1,
  },
  "status": "FINISHED",
}
`;
